app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -776,67 +766,52 @@ private void unregisterBroadcastReceiver() {
     //////////////////////////////////////////////////////////////////////////*/
     //region Thumbnail loading
 
-    private Target getCurrentThumbnailTarget() {
-        // a Picasso target is just a listener for thumbnail loading events
-        return new Target() {
-            @Override
-            public void onBitmapLoaded(final Bitmap bitmap, final Picasso.LoadedFrom from) {
-                if (DEBUG) {
-                    Log.d(TAG, "Thumbnail - onBitmapLoaded() called with: bitmap = [" + bitmap
-                            + " -> " + bitmap.getWidth() + "x" + bitmap.getHeight() + "], from = ["
-                            + from + "]");
-                }
-                // there is a new thumbnail, so e.g. the end screen thumbnail needs to change, too.
-                onThumbnailLoaded(bitmap);
-            }
-
-            @Override
-            public void onBitmapFailed(final Exception e, final Drawable errorDrawable) {
-                Log.e(TAG, "Thumbnail - onBitmapFailed() called", e);
-                // there is a new thumbnail, so e.g. the end screen thumbnail needs to change, too.
-                onThumbnailLoaded(null);
-            }
-
-            @Override
-            public void onPrepareLoad(final Drawable placeHolderDrawable) {
-                if (DEBUG) {
-                    Log.d(TAG, "Thumbnail - onPrepareLoad() called");
-                }
-            }
-        };
-    }
-
     private void loadCurrentThumbnail(final List<Image> thumbnails) {
         if (DEBUG) {
             Log.d(TAG, "Thumbnail - loadCurrentThumbnail() called with thumbnails = ["
                     + thumbnails.size() + "]");
         }
 
-        // first cancel any previous loading
-        cancelLoadingCurrentThumbnail();
-
         // Unset currentThumbnail, since it is now outdated. This ensures it is not used in media
-        // session metadata while the new thumbnail is being loaded by Picasso.
+        // session metadata while the new thumbnail is being loaded by Coil.
         onThumbnailLoaded(null);
         if (thumbnails.isEmpty()) {
             return;
         }
 
         // scale down the notification thumbnail for performance
-        PicassoHelper.loadScaledDownThumbnail(context, thumbnails)
-                .tag(PICASSO_PLAYER_THUMBNAIL_TAG)
-                .into(currentThumbnailTarget);
-    }
+        final var target = new Target() {
```

Comment:
Shouldn't the target be stored somewhere to avoid it being garbage collected?

Reply:
The enqueue method holds a strong reference to the target: https://github.com/coil-kt/coil/blob/ce14e14bc9a25c6952e26e248aad75fbd2e9f9d0/coil-base/src/main/java/coil/RealImageLoader.kt#L113